lib/coelho/connection.ex

```diff
@@ -6,21 +6,24 @@ defmodule Coelho.Connection do
   @reconnect_interval_ms 5000
 
   def init(_opts) do
-    {:ok, %{}, 0}
+    {:ok, new_state(), 0}
   end
 
   def start_link() do
     GenServer.start_link(__MODULE__, %{}, name: __MODULE__)
   end
 
-  def connect() do
+  def connect(state) do
     config = Application.get_env(:coelho, Coelho.Connection)
 
     case AMQP.Connection.open(config) do
       {:ok, conn} ->
         Logger.debug("Connected to rabbitmq: #{inspect(conn)}")
-        Process.monitor(conn.pid)
-        {:ok, conn}
+        conn_ref = Process.monitor(conn.pid)
```

Comment:
`amqp.connection.open` não esta' linkado no processo? se for linkado, p/ esse monitor funcionar vc teria q ligar `:trap_exit`.

Reply:
Não é _linkado_ mesmo. :sake: